File: rest/kotlin/ktor/src/commonMain/kotlin/m4gshm/benchmark/model/Task.kt
package m4gshm.benchmark.model

import kotlinx.datetime.Instant
import kotlinx.serialization.Serializable

@Serializable
data class Task(
    val id: String? = null, val text: String? = null, val tags: List<String>? = null,
    val deadline: Instant? = null
)

File: rest/go/model/task.go
package model

import (
	"benchmark/rest/storage"
	"time"
)

//go:generate fieldr -type Task --debug
//go:fieldr new-full
//go:fieldr get-set

type Task struct {
	ID       string     `json:"id,omitempty"`
	Text     string     `json:"text,omitempty"`
	Tags     []string   `json:"tags,omitempty"`
	Deadline *time.Time `json:"deadline,omitempty" format:"date-time"`
}

var _ storage.IDAware[string] = (*Task)(nil)
